hack/verify-lint.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+# standard bash error handling
+set -o nounset # treat unset variables as an error and exit immediately.
+set -o errexit # exit immediately when a command fails.
+set -E         # needs to be set if we want the ERR trap
+
+readonly CURRENT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+readonly ROOT_PATH="${1:-$( cd "${CURRENT_DIR}/.." && pwd )}" # first argument or root of the project
+readonly TMP_DIR=$(mktemp -d)
+readonly GOLANGCI_LINT_VERSION="v1.38.0"
+
+source "${CURRENT_DIR}/utilities.sh" || { echo 'Cannot load CI utilities.'; exit 1; }
+
+cleanup() {
+    rm -rf "${TMP_DIR}" || true
+}
+
+trap cleanup EXIT SIGINT
+
+golangci::install() {
+  export PATH="${INSTALL_DIR}:${PATH}"
+
+  shout "Install the golangci-lint in version ${GOLANGCI_LINT_VERSION}"
+  curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b "${INSTALL_DIR}" ${GOLANGCI_LINT_VERSION}
+  echo -e "${GREEN}√ install golangci-lint${NC}"
+
+}
+
+golangci::run_checks() {
+  shout "Run golangci-lint checks"
+  LINTS=(
+    # default golangci-lint lints
+    deadcode errcheck gosimple govet ineffassign staticcheck \
+    structcheck typecheck unused varcheck \
+    # additional lints
+    golint gofmt misspell gochecknoinits unparam scopelint gosec
+  )
+
+  echo "Checks: ${LINTS[*]}"
+  cd $ROOT_PATH
+  golangci-lint --disable-all --enable="$(sed 's/ /,/g' <<< "${LINTS[@]}")" --timeout=10m run --config $CURRENT_DIR/.golangci.yml
```

Comment:
Can we remove this ugly sed and just use `,` in `LINTS`?

Reply:
Generally we can get rid of this script and put the execution of lint in makefile.

Reply:
The idea here was that `LINTS` is easy to read by a human so that we can:
- Easily see/add/remove which linters we are running.
- Document why we have specific linters.

I thought of having this in the `Makefile` but found it useful to be able to have this working out of the box without previous setup and more detailed output.